midterm_project/vlad_butulan/shopping_list.py

```diff
@@ -0,0 +1,205 @@
+import random
+import pprint
+from collections.abc import Mapping, MutableSequence, MutableMapping
+
+pp = pprint.PrettyPrinter(indent=4)
+
+pp_list_header = r"""
+   ______________________________
+ / \                             \.
+|   |     Shopping list:         |.
+ \_ |                            |.
+    |                            |.
+"""
+
+pp_list_tailer = r"""
+    |                            |.
+    |   _________________________|___
+    |  /                            /.
+    \_/vb__________________________/.
+    """
+
+shopping_list_archive = []
+
+
+class PrittyPrinter():
+    """a mixin class used by Fridge and Recipe to print nicely their contents"""
+    
+    def pp_print_recipe(self):
+        global pp
```

Comment:
I think you don't need `global`, `pp` will be accessible

Reply:
fixed in v1.1